### ISSUE TITLE
pcd_demo.py - error - list indices must be integers 

### ISSUE THREAD

**Issue by mihsamusev** [open-mmlab/mmdetection3d#115]
Thanks for you work, im trying to run `pcd_demo.py` exactly as described in [getting started](https://github.com/open-mmlab/mmdetection3d/blob/master/docs/getting_started.md), using KITTI config and a latest SECOND-KITTI-3d-Car checkpoint that i got [here](https://github.com/open-mmlab/mmdetection3d/tree/master/configs/second) 

i run it using the following shell script:
```
python demo/pcd_demo.py demo/kitti_000008.bin \
configs/second/hv_second_secfpn_6x8_80e_kitti-3d-car.py \
checkpoints/hv_second_secfpn_6x8_80e_kitti-3d-car_20200620_230238-393f000c.pth
```
Result:

```
Traceback (most recent call last):
  File "demo/pcd_demo.py", line 28, in <module>
    main()
  File "demo/pcd_demo.py", line 24, in main
    show_result_meshlab(data, result, args.out_dir)
  File "/home/msa/Documents/repos/mmdetection3d/mmdetection3d/mmdet3d/apis/inference.py", line 102, in show_result_meshlab
    pred_bboxes = result['boxes_3d'].tensor.numpy()
TypeError: list indices must be integers or slices, not str
```

I also get exact same results if i try run demo with pointpillars-3d-car config / chekcpoint.
**Environment**

1. `python mmdet3d/utils/collect_env.py` output:
```
sys.platform: linux
Python: 3.7.9 (default, Aug 31 2020, 12:42:55) [GCC 7.3.0]
CUDA available: True
GPU 0: Quadro M2200
CUDA_HOME: /usr
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GCC: gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
PyTorch: 1.6.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.7.0
OpenCV: 4.4.0
MMCV: 1.1.2
MMCV Compiler: GCC 9.3
MMCV CUDA Compiler: 10.1
MMDetection: 2.4.0
MMDetection3D: 0.5.0+1312430
```
2. Error happens at mmde3d.apis show_results_meshlab so maybe its version is relevant:
```
pip list | grep mmdet
```
outputs

```
mmdet                  2.4.0
mmdet3d                0.5.0
```

**Comment by ZwwWayne** [comment]
This issue might be caused by the migration of batch inference. @xavierwu95 please have a check of the inference API.

**Comment by mihsamusev** [comment]
@ZwwWayne thanks for fast response, can you give some tips how to do it? Im not really sure what you mean.

**Comment by ZwwWayne** [comment]
Fixed by #116
